test/parallel-benchmark/mzcompose.py

```diff
@@ -0,0 +1,1190 @@
+#                     """
+#                 ),
+#                 # TODO: Inserts, Selects
+#                 LoadPhase(
+#                     duration=120,
+#                     actions=[],
+#                 ),
+#             ]
+#         )
+
+
+class Statistics:
+    def __init__(self, times: list[float], values: list[float]):
+        assert len(times) == len(values)
+        self.queries: int = len(times)
+        self.qps: float = len(times) / max(times)
+        self.max: float = max(values)
+        self.min: float = min(values)
+        self.avg = numpy.mean(values)
+        self.p50 = numpy.median(values)
+        self.p95 = numpy.percentile(values, 95)
+        self.p99 = numpy.percentile(values, 99)
+        self.std = numpy.std(values, ddof=1)
+
+    def __str__(self) -> str:
+        return f"""  queries: {self.queries:>5}
+  qps: {self.qps:>7.2f}
+  max: {self.max:>7.2f}ms
+  min: {self.min:>7.2f}ms
+  avg: {self.avg:>7.2f}ms
+  p50: {self.p50:>7.2f}ms
+  p95: {self.p95:>7.2f}ms
+  p99: {self.p99:>7.2f}ms
+  std: {self.std:>7.2f}ms"""
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    c.silent = True
+
+    parser.add_argument(
+        "--redpanda",
+        action="store_true",
+        help="run against Redpanda instead of the Confluent Platform",
+    )
+
+    parser.add_argument(
+        "--guarantees",
+        action="store_true",
+        default=True,
+        help="Check guarantees defined by test scenarios",
+    )
+
+    parser.add_argument(
+        "--size",
+        metavar="N",
+        type=int,
+        default=4,
+        help="default SIZE",
+    )
+
+    parser.add_argument(
+        "--scenario", metavar="SCENARIO", type=str, help="Scenario to run."
+    )
+
+    parser.add_argument("--mz-url", type=str, help="Remote Mz instance to run against")
+
+    args = parser.parse_args()
+
+    if args.scenario:
+        assert args.scenario in globals(), f"scenario {args.scenario} does not exist"
+        scenarios = [globals()[args.scenario]]
+    else:
+        scenarios = all_subclasses(Scenario)
+
+    service_names = ["materialized", "postgres", "mysql"] + (
+        ["redpanda"] if args.redpanda else ["zookeeper", "kafka", "schema-registry"]
+    )
+
+    failures: list[TestFailureDetails] = []
+
+    if args.mz_url:
+        with c.override(
+            Testdrive(
+                no_reset=True,
+                materialize_url=args.mz_url,
+                seed=1,
+                no_consistency_checks=True,
+            )
+        ):
+            target = parse_pg_conn_string(args.mz_url)
+            c.up("testdrive", persistent=True)
+            conn_infos = {"materialized": target}
+            conn = target.connect()
+            with conn.cursor() as cur:
+                mz_version = cur.execute("SELECT mz_version()")
+                mz_version = cur.fetchall()[0][0]
+            conn.close()
+            mz_string = f"{mz_version} ({target.host})"
+    else:
```

Comment:
What's the comparison target in this case? Or is no comparison performed?

Reply:
No comparison performed, only checks the guarantees.